### PR TITLE
201 - Adding label next to discussion badges

### PR DESCRIPTION
**Pull request by smithandre**
Adding labels next to the discussion badges, addressing https://github.com/flarum/core/issues/201

### CODE REVIEW

File: js/lib/components/Badge.js [hunk 1]
```diff
@@ -25,9 +25,12 @@ export default class Badge extends Component {
...
     return (
-      <span {...attrs}>
-        {iconName ? icon(iconName, {className: 'Badge-icon'}) : m.trust('&nbsp;')}
-      </span>
+        <div>
+          <span {...attrs}>
+            {iconName ? icon(iconName, {className: 'Badge-icon'}) : m.trust('&nbsp;')}
+          </span>
+          <span className="Badge-label">{this.props.label}</span>
+        </div>
```
Comment (franzliedke) [inline js/lib/components/Badge.js]: Should this wrapping div receive some styling to prevent wrapping between the two spans?

Comment (franzliedke) [reply js/lib/components/Badge.js]: Also, it looks like the indentation changed? Was that on purpose? If not, please revert.

**Comment by franzliedke** [comment]
When you change any of our JS code, please make sure you also recompile and commit the code in the district directory.

**Comment by franzliedke** [comment]
*dist directory

File: js/lib/components/Badge.js (hunk 1 above)
Comment (tobscure) [reply js/lib/components/Badge.js]: Tthe wrapping tag should be a `span` as well because badges are inline elements.

**Comment by josephnle** [comment]
@franzliedke Just curious and wanted to clarify since I didn't do this for my PR. The [Contribution Guide](https://github.com/flarum/flarum/blob/master/CONTRIBUTING.md#development-setup) tells us not to commit compiled assets. Should we start committing the compiled JS?

**Comment by franzliedke** [comment]
Uhm, looks like we need to update that guide, right, @tobscure. At least that has been our policy for the last couple of months.

Good catch, @josephnle.

Pleaes take care of the requested wrapping `<span>`. If you don't commit the compiled JS, we can do it after the merge.

**Comment by franzliedke** [comment]
Indeed, I was wrong. We take care of the compilation ourselves, in order to minimize conflicts between different pull requests.

@josephnle Still interested in finishing this? Otherwise, we will have to take care of the rest. :smile:

**Comment by josephnle** [comment]
@franzliedke Would be happy to help out and finish this, but this isn't my PR. 😛 Perhaps you meant to tag @smithandre?

**Comment by franzliedke** [comment]
Oh, haha. Yes, that question was for @smithandre. :grinning:

File: js/lib/components/Badge.js (hunk 1 above)
Comment (franzliedke) [reply js/lib/components/Badge.js]: Do we need to return exactly one object from the `view` method or could we return multiple as well, by the way?

**Comment by franzliedke** [comment]
@tobscure I'll let you finish this. :grin:

**Comment by franzliedke** [comment]
Closing this, because reviews weren't followed up.

**Pull request closed without merging**